src/error.rs

```diff
@@ -300,6 +314,22 @@ impl fmt::Display for Error {
 #[cfg(feature = "std")]
 impl ::std::error::Error for Error {}
 
+/// Additional context for the `CertNotValidForName` error variant.
+///
+/// The contents of this type depend on whether the `alloc` feature is enabled.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct InvalidNameContext {
+    /// Expected server name.
+    #[cfg(feature = "alloc")]
+    pub expected: ServerName<'static>,
+    /// The names presented in the end entity certificate.
+    ///
+    /// Unlike `expected`, these names might contain wildcard labels, and can contain both DNS
```

Comment:
Will `expected` be something a consumer of this API will understand, or is it an internal detail that's leaking here?

Maybe something like "These are the subject names as present in the certificate and may contain IP address names, and DNS names with a wildcard label"?

Reply:
Adapted your proposal, what do you think? Maybe we should also reword the `expected` docstring to have more context for users?

Reply:
The update looks good to me. I wasn't able to think of any more meaningful context.